src/components/Api.js

```diff
@@ -0,0 +1,56 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _request(urlEnd, options) {
+    return fetch(this._baseUrl+urlEnd, options)
+    .then(this._handleRequest)
+  }
+
+  _handleRequest(res) {
+    if (!res.ok) {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+    return res.json();
+  }
+
+  getUserInfo(endUrl, options) {
+    return this._request(endUrl,options);
+    
+  }
+
+  getInitialCards(endUrl, options) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    })
+      .then(this._handleRequest);
+  }
+
+  deleteCard(endUrl, options) {
```

Comment:
Please, don't pass `endUrl` into each method from `index.js`. It should be done inside `Api.js` to clean up the other files of the project


```js
  deleteCard(cardId) {
    const endUrl = "/cards/" + cardId;
    const options = {
      method: "DELETE",
      headers: headers,
    };
    return this._request(endUrl, options);
  }

```